resend.go

```diff
@@ -80,6 +87,26 @@ func NewCustomClient(httpClient *http.Client, apiKey string) *Client {
 	return c
 }
 
+// NewRequestWithOptions builds and returns a new HTTP request object
+// based on the given arguments and options
+// It is used to set additional options like idempotency key
+func (c *Client) NewRequestWithOptions(ctx context.Context, method, path string, params interface{}, options Options) (*http.Request, error) {
+	req, err := c.NewRequest(ctx, method, path, params)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Set the idempotency key if provided and the method is POST for now.
+	if options != nil {
```

Comment:
I'm not sure if we will add idempotency key to all post methods, but I think this is ok for now